tck/src/main/java/org/eclipse/microprofile/lra/tck/TckLRATypeTests.java

```diff
@@ -0,0 +1,349 @@
+     *                     by the {@link LRA.Type} annotation
+     * @param methodLRAShouldBeActive if true the LRA started by the invoked method should still
+     *                               be active after the resource invocation completes
+     */
+    private void resourceRequest(String path, boolean startLRA, int expectedStatus, MethodLRACheck lraCheckType,
+                                 boolean methodLRAShouldBeActive) {
+        Invocation.Builder target = tckSuiteTarget.path(TCK_LRA_TYPE_RESOURCE_PATH)
+                .path(path).request();
+        URI lra = startLRA ? lraClient.startLRA(null, lraClientId(), lraTimeout(), ChronoUnit.MILLIS) : null;
+
+        Response response = lra == null ? target.get() : target.header(LRA.LRA_HTTP_HEADER, lra).get();
+
+        try {
+            String methodLRA = response.readEntity(String.class);
+            String incomingLRA = lra == null ? "" : lra.toASCIIString();
+
+            assertEquals(testName.getMethodName() + ": Unexpected status", expectedStatus, response.getStatus());
+
+            if (response.getStatus() == Response.Status.PRECONDITION_FAILED.getStatusCode()) {
+                // 412 errors should abort running the target method so skip the LRA check
+                lraCheckType = MethodLRACheck.NONE;
+                methodLRAShouldBeActive = false;
+            }
+
+            switch (lraCheckType) {
+                case NOT_PRESENT:
+                    assertEquals(testName.getMethodName() + ": Resource method should not have run with an LRA: " + methodLRA,
+                            0, methodLRA.length());
+                    break;
+                case EQUALS:
+                    assertEquals(testName.getMethodName() + ": Resource method should have ran with the incoming LRA",
+                            incomingLRA, methodLRA);
+                    break;
+                case NOT_EQUALS:
+                    assertNotEquals(testName.getMethodName() + ": Resource method should not have run with the incoming LRA",
+                            incomingLRA, methodLRA);
+                    break;
+                default:
+                    break;
+            }
+
+            if (methodLRAShouldBeActive) {
+                // validate that the method ran with an LRA and that it is still active
+                assertNotEquals(testName.getMethodName() + ": Resource method should not have run with an LRA: " + methodLRA,
+                        0, methodLRA.length());
+                assertFalse(lraClient.isLRAFinished(methodLRA));
+                lraClient.closeLRA(methodLRA);
+            } else if (methodLRA.length() != 0) {
+                // otherwise it should be finished
+                assertTrue(lraClient.isLRAFinished(methodLRA));
+            }
+
+            if (lra != null) {
+                lraClient.closeLRA(lra);
+            }
+        } finally {
+            response.close();
```

Comment:
could not be the `response` `null` in some bad circumstances

Reply:
The response object is initialised on line 285 and it is outside of the try/finally block so if that failed then an exception would be thrown and the thread would never have entered the try/finally block, i.e. it cannot be null:

> Response response = lra == null ? target.get() : target.header(LRA.LRA_HTTP_HEADER, lra).get();
> try {
> ...